sdk/tests/conformance2/extensions/ext-color-buffer-float.html

```diff
@@ -270,6 +296,41 @@
     shouldBe('gl.getExtension("EXT_color_buffer_float").myProperty', '2');
 }
 
+function runInternalFormatQueryTest()
+{
+    debug("");
+    debug("testing the internal format query");
+
+    var maxSamples = gl.getParameter(gl.MAX_SAMPLES);
+    var formats = new Array(gl.RGBA16F, gl.R32F, gl.RG32F, gl.RGBA32F, gl.R16F, gl.RG16F, gl.R11F_G11F_B10F);
+    var firstMultiOnlyFormat = 4;
+    for (var fmt = 0; fmt < formats.length; ++fmt) {
+        var samples = gl.getInternalformatParameter(gl.RENDERBUFFER, formats[fmt], gl.SAMPLES);
+        if (fmt >= firstMultiOnlyFormat && (samples.length == 0 || samples[0] < maxSamples)) {
+            testFailed("the maximum value in SAMPLES should be at least " + maxSamples);
+            return;
+        }
+
+        var prevSampleCount = 0;
+        var sampleCount;
+        for (var ndx = 0; ndx < samples.length; ++ndx, prevSampleCount = sampleCount) {
+            sampleCount = samples[ndx];
+            // sample count must be > 0
```

Comment:
"If internalformat is GL_RGBA16F, GL_R32F, GL_RG32F, or GL_RGBA32F, the value of **GL_NUM_SAMPLE_COUNTS may be zero**, or else the maximum value in GL_SAMPLES may be less than the value of GL_MAX_SAMPLES"

I think this mostly indicates that the length of returned array, not its elements may zero. Actually the elements should be positive according to the below sentence.

"if pname is SAMPLES, the sample counts supported for internalformat and target are written into params, in descending numeric order. **Only positive values are returned.**"